src/server/typingsInstaller/nodeTypingsInstaller.ts

```diff
@@ -63,12 +82,27 @@ namespace ts.server.typingsInstaller {
                 this.log.writeLine(`Process id: ${process.pid}`);
             }
             this.npmPath = getNPMLocation(process.argv[0]);
-            this.exec = require("child_process").exec;
-            this.httpGet = require("http").get;
+            let execSync: ExecSync;
+            ({ exec: this.exec, execSync } = require("child_process"));
+
+            this.ensurePackageDirectoryExists(globalTypingsCacheLocation);
+
+            try {
+                if (this.log.isEnabled()) {
+                    this.log.writeLine(`Updating ${TypesRegistryPackageName} npm package...`);
+                }
+                execSync(`${this.npmPath} install ${TypesRegistryPackageName}`, { cwd: globalTypingsCacheLocation, stdio: "ignore" });

```

Comment:
Right, that's what I meant (updated my comment after I wrote it), but still - it does mean it may not receive commands like "close" for a while. I assume there's no/minimal harm in having one linger a little if tsserver restarts and spawns another before it closes?